executor/load_data.go

```diff
@@ -168,6 +240,34 @@ func (e *LoadDataInfo) getValidData(prevData, curData []byte) ([]byte, []byte) {
 	return nil, curData
 }
 
+func (e *LoadDataInfo) insertDataWithCommit(ctx context.Context, prevData, curData []byte) ([]byte, error) {
+	var err error
+	var reachLimit bool
+	for {
+		prevData, reachLimit, err = e.InsertData(ctx, prevData, curData)
+		if err != nil {
+			return nil, err
+		}
+		if !reachLimit {
+			break
+		}
+		err := e.CheckAndInsertOneBatch(ctx)
+		if err != nil {
+			return nil, err
```

Comment:
`StmtRollback()` ?

When you call StmtCommit, you should always consider the StmtRollback, otherwise there may be dirty data remain in the session and pollute the next statement in this session!

Reply:
the caller of this will process error and rollback, originally in the clientConn.handleQuery -> handleLoadData logic, here in the end of DoCmd

Reply:
`Rollback` and `StmtRollback` is not the same thing.